A62-pick-first.md

```diff
@@ -0,0 +1,148 @@
+an aggregate cluster, and where the `LOGICAL_DNS` cluster is not the last
+cluster in the list. Each cluster under the aggregate cluster is represented as
+a child policy under `priority`, and the leaf policy for a `LOGICAL_DNS` cluster
+is `pick_first`. Without sticky-TF support in `pick_first`, it can lead to a
+situation where the `priority` LB policy continues to send picks to a higher
+priority `LOGICAL_DNS` cluster when none of the addresses behind it are
+reachable, because `pick_first` doesn't report `TRANSIENT_FAILURE` as it's
+connectivity state. See [gRFC A37][A37] for more details on aggregate clusters.
+
+Providing sticky-TF in all `pick_first` implementations would enable us to
+overcome these shortcomings. This would involve making the following changes to
+`pick_first` implementations, once connections to all addresses fail:
+- Report `TRANSIENT_FAILURE` as the connectivity state.
+- Attempt to reconnect to the addresses indefinitely until a connection succeeds
+  (at which point, they should report `READY`), or there is no RPC activity on
+  the channel for the specified `IDLE_TIMEOUT`.
+
+
+All gRPC implementations should implement `IDLE_TIMEOUT` and have it enabled by
+default. A default value of 30 minutes is recommended.
+
+[A56]: A56-priority-lb-policy.md
+[A37]: A37-xds-aggregate-and-logical-dns-clusters.md
+
+### Enable random shuffling of address list
+
+Our general philosophy is that the address order is to be determined by the name
+resolver server, or by the name resolver client performing sorting as described
+in [RFC-8304 section 4](https://www.rfc-editor.org/rfc/rfc8305#section-4). In
+some DNS configurations though, all clients get the addresses in the same order,
+either because the authoritative server does that or because of caching. In
+these scenarios, it is desirable for `pick_first` to support random shuffling of
+received addresses to provide better load balancing.
+
+
+At the time of this writing, `pick_first` implementations do not expect any
+configuration to be passed to it. As part of this design, we will add a field to
+its configuration that would enable it to randomly shuffle the order of the
+addresses it receives before it starts to attempt to connect.
+
+```
+{
+    // If set to true, instructs the LB policy to shuffle the order of the
+    // list of addresses received from the name resolver before attempting to
+    // connect to them.
+    "shuffleAddressList": boolean
+}
+```
+
+gRPC clients that receive `pick_first` configuration with this field set, but do
+not support this configuration should continue to work and their behavior should
+remain unchanged. Implementations that support the new field and receive
+configuration with `shuffleAddressList` set to `true`, should shuffle the
+received address list at random before attempting to connect to them.
+
+### Temporary environment variable protection
+
+During initial development, random shuffling of address list will be enabled by
```

Comment:
Please also note that the env var will also guard accepting the `pick_first` config via xDS.

Reply:
Done. Also added A52 to the related proposals list and linked it from here.